qiskit_sphinx_theme/__init__.py

```diff
@@ -6,14 +6,108 @@
+QISKIT_ORG_LINK = {
+    'label': 'Qiskit home page',
+    'url': QISKIT_ORG
+}
+
+TOP_MENU_LINKS = {
+    'qiskit_core_getting_started': {
+        'label': 'Getting Started',
+        'path': 'getting_started'
+    },
+    'qiskit_core_tutorials': {
+        'label': 'Tutorials',
+        'path': 'tutorials'
+    },
+    'qiskit_partners': {
+        'label': 'Partners',
+        'url': QISKIT_PARTNERS_DOCS,
+    },
+    'qiskit_applications': {
+        'label': 'Applications',
+        'children': [
+            {
+                'label': 'Machine learning',
+                'url': QISKIT_MACHINE_LEARNING_DOCS,
+                'description': 'QSVM, VQC (Variational Quantum Classifier), and QGAN (Quantum Generative Adversarial Network) algorithms.'
+            },
+            {
+                'label': 'Nature',
+                'url': QISKIT_NATURE_DOCS,
+                'description': 'Quantum applications in chemistry, physics, and biology.'
+            },
+            {
+                'label': 'Finance',
+                'url': QISKIT_FINANCE_DOCS,
+                'description': 'Uncertainty components for stock/securities problems, Ising translators for portfolio optimizations and data providers to source real or random data.'
+            },
+            {
+                'label': 'Optimization',
+                'url': QISKIT_OPTIMIZATION_DOCS,
+                'description': 'High-level optimization problems that are ready to run on simulators and real quantum devices'
+            }
+        ]
+    },
+    'qiskit_experiments': {
+        'label': 'Experiments',
+        'url': QISKIT_EXPERIMENTS_DOCS,
+    },
+    'resources': {
+        'label': 'Resources',
+        'children': [
+            {
+                'label': 'Slack support',
+                'url': QISKIT_SLACK
+            },
+            {
+                'label': 'Qiskit Textbook',
+                'url': QISKIT_TEXTBOOK
+            },
+            {
+                'label': 'Qiskit events',
+                'url': QISKIT_EVENTS
+            }
+        ]
+    },
+    'qiskit_organization_github': {
+        'label': 'GitHub',
+        'url': QISKIT_ORGANIZATION_GITHUB
+    }
+}
 
 def get_html_theme_path():
     """Return list of HTML theme paths."""
     cur_dir = path.abspath(path.dirname(path.dirname(__file__)))
     return cur_dir
 
-# See http://www.sphinx-doc.org/en/stable/theming.html#distribute-your-theme-as-a-python-package
+def map_to_theme_links(app, config):
+    """Map the theme options given in the config file to links"""
+    theme_options = config.html_theme_options or {}
+
+    theme_top_menu_links = list(map(lambda top_menu_link: TOP_MENU_LINKS[top_menu_link], theme_options.get('top_menu_links')))
+    config.html_theme_options['top_menu_links'] = theme_top_menu_links
+
+def include_additional_theme_links(app, config):
+    """Include other useful links on the theme"""
+    config.html_theme_options['qiskit_org_link'] = QISKIT_ORG_LINK
```

Comment:
One of the goals behind this refactor is the no need to set links on `conf.py` to have just one source of truth for the links. For example, one of the scenarios that we have right now: the textbook link changed from the old textbook to the new one. This link was set by hand and individually on every repo, so to have the correct and same link through all the documentation, we need to update all the repos. With just one source of truth we can update documentation much faster, be more consistent and find errors faster.

Once we have this refactor working on the different repos, we can update the README file (or the contributing guide) to reflect how to work with this structure. Do you think this will be enough?

Reply:
I wrote more in the other thread, but I think this PR actually makes the packages need to do _more_ to define the links than they currently need to do - neither the metapackage nor Terra currently define _anything_ about what the top bar should look like in their `conf.py`, but after this PR, they'll need to include a new variable `top_menu_links = ` that have the same content.

(Just to be clear: `conf.py` in this repository defines settings for the documentation build that other repositories will not inherit.  If the PR needs to add stuff in `conf.py`, _every_ downstream package will need to duplicate those additions.)

Reply:
@y4izus Jake has just summarised what I meant when I was talking to you a minute ago 😅 As we discussed I think the better solution is to create a feature branch for the top menu changes, move this PR over to there and only merge the completed version of the top menu into main (i.e. when we have a proper design and a standard set of menu links for all packages). Then we can get the packages to update their docs to pull in the finalised version instead of this intermediate phase.

@jakelishman to clarify, the changes you see here are meant to be an intermediate step in a longer process of updating the menu bar. Step 1 here is to make the top menu bar into kind of a reusable component and remove some of the need for the packages to completely make their own top menu from scratch (which they all currently do in `layout.html`). You're right that they previously would not have to do anything in their `conf.py` and now they do, but they would no longer need to manage their own menu bar html and corresponding `theme_variables`. Step 2 would be to remove any need for them to specify `top_menu_links` at all, and have a single uniform top menu for all packages, with all the links and html managed in the sphinx theme repo (i.e. have a single source off truth for the menu for all packages). Step 1 is needed here to enable Yaiza and others to be able to start working on  the structure of the unified top menu bar while still waiting for some design work to be done on a final version. The packages themselves don't currently use the sphinx theme at all for the top menu bar and theres no requirement yet for them to make any changes. And anyway I think we shouldn't get the packages to change anything yet until we have completed unified menu bar, hence why moving this ongoing menu bar work into a feature branch is probably the best way forward with this. Then once we have a completed unified top menu bar we can set out a strategy for updating each of the packages to pull it in with minimal changes needed on their part

Reply:
Ok, that plan sounds good to me, I think I understand what's going on now. I'm really happy that we're looking at the theme, and working out how to do things properly!

Reply:
Talking with @javabster I realized that I didn't explain very well the intermediate step, sorry for that @jakelishman . In addition I was thinking on doing this intermediate step without bother the different teams, but it's true that we can work on a feature branch and reduce some work. I'm waiting for permissions to create that branch, until then, I put the PR as draft. Thank you both for your feedback!